fastlane_core/lib/fastlane_core/build_watcher.rb

```diff
@@ -23,6 +23,7 @@ def wait_for_build_processing_to_be_complete(app_id: nil, platform: nil, train_v
         platform = Spaceship::ConnectAPI::Platform.map(platform) if platform
         UI.message("Waiting for processing on... app_id: #{app_id}, app_version: #{app_version}, build_version: #{build_version}, platform: #{platform}")
 
+        build_watching_start_time = Time.new
```

Comment:
Please read my other comment first about changing the `message` to `verbose` in the loop

Maybe we print the estimate timeout time here if `timeout_duration` is given instead of printing it out every time in the loop?

```suggestion
        build_watching_start_time = Time.new
        unless timeout_duration.nil?
          end_time = build_watching_start_time + timeout_duration
          UI.message("Will timeout watching build after #{timeout_duration} seconds around #{end_time} time!")
        end
```

Note: ☝️ I did not test this but maybe something like this? 🤷‍♂️

Reply:
Added Unit test so test this logic, every time on CI, btw this logic was correct! 😉